_posts/2017/2017-10-15-rescue-specific-errors-avoid-standarderror-do-not-rescue-exception.md

```diff
@@ -0,0 +1,63 @@
+---
+title: 'Rescue specific errors. Avoid rescuing StandardError. Don’t rescue Exception.'
+description: 'Rescuing all the things is bad.'
+layout: article
+category: ruby
+image:
+  base: '2017/rescue-specific-errors-avoid-standarderror-do-not-rescue-exception'
+  alt: 'David Hasslehoff. Smouldering.'
+---
+
+There are many built-in error classes in Ruby and Rails. Most of these errors are subclasses of Ruby’s StandardError. You can find more information in the relevant [Ruby docs](http://ruby-doc.org/core-2.4.2/StandardError.html).
+
+## Instead of…
+
+…rescuing `Exception`.
+
+```ruby
+def your_method
+  # do something
+rescue Exception => e
+  # saved ALL THE THINGS
+end
+```
+
+## Or…
+
+…a non-specific `rescue` that implicitly rescues `StandardError`.
+
+```ruby
+def your_method
+  # do something
+rescue => e
+  # saved StandardError and all subclasses
+end
+```
+
+### Or, much better…
```

Comment:
...or this one to 'Use'..because at the moment you have a sentence that reads: "Instead of...Or...Or, much better" but you want either "Instead of...Use...Or, much better..." or "Instead of...Or...Use...".

Reply:
Good catch. This should be ‘Use’